### PR TITLE
refactor: make ref checkout for picasso scripts configurable

### PR DESCRIPTION
**Pull request by mariajgrimaldi**
### Description

This change makes the use of versions different than default possible, so the code available matches the version of the workflow used.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -34,6 +34,11 @@ on:
         required: false
         type: string
         default: '3.12'
+      PICASSO_VERSION:
```
Comment (BryanttV) [inline .github/workflows/build.yml:37]: Should we document this new input?

Comment (mariajgrimaldi) [reply .github/workflows/build.yml:37]: Yes, we should! Thank you.

File: docs/source/topic_guides/reusable_build_workflow.rst [hunk 2]
```diff
@@ -22,14 +22,15 @@ The Picasso Workflow is designed to build Open edX Docker images for Tutor envir
...
 - **Private repository access**: SSH keys are used to clone private repositories securely. The SSH private key should be stored as a secret in the repository, and must have access to the repository specified in ``STRAIN_REPOSITORY``.
+- **Configures docker registry**: The workflow sets up the Docker registry to push images to Dockerhub or AWS ECR.
 - **Extra commands**: The workflow allows running additional custom commands with ``tutor picasso run-extra-commands``. For details, refer to the `tutor-contrib-picasso`_ documentation.
-- **Environment setup**: The workflow sets up and configures Tutor Virtual Environments (TVM), installs necessary plugins like ``tutor-contrib-picasso``, and prepares the environment to build and push Docker images using the `Tutor CLI`_.
+- **Environment setup**: The workflow sets up installs necessary plugins like ``tutor-contrib-picasso``, and prepares the environment to build and push Docker images using the `Tutor CLI`_.
```
Comment (BryanttV) [inline docs/source/topic_guides/reusable_build_workflow.rst:33]: The `README` still mentions TVM:
https://github.com/eduNEXT/picasso/blob/199bb1aeaf0e79f72cfdcbb137e9c3cb3cdcab73/README.rst?plain=1#L29

File: docs/source/how_tos/configure_workflow.rst [hunk 3]
```diff
@@ -34,6 +37,9 @@ Here is an example of how to use the Picasso workflow with the inputs set:
                 STRAIN_PATH: redwood/base
                 SERVICE: mfe
                 ENABLE_LIMIT_BUILDKIT_PARALLELISM: false
+                RUNNER_WORKFLOW_LABEL: ubuntu-latest
```
Comment (BryanttV) [inline docs/source/how_tos/configure_workflow.rst:40]: Should we include these inputs in the README?

**Review by magajh — APPROVED** [review]

File: docs/source/topic_guides/reusable_build_workflow.rst (hunk 2 above)
Comment (mariajgrimaldi) [reply docs/source/topic_guides/reusable_build_workflow.rst:33]: Done, thank you!

File: docs/source/how_tos/configure_workflow.rst (hunk 3 above)
Comment (mariajgrimaldi) [reply docs/source/how_tos/configure_workflow.rst:40]: I did that here: https://github.com/eduNEXT/picasso/pull/37/commits/f186a2cb84b7d18b3d6759a141cdf2eb8220dc0a#diff-7b3ed02bc73dc06b7db906cf97aa91dec2b2eb21f2d92bc5caa761df5bbc168fR91, thank you!

**Pull request merged**